client/keystore/src/local.rs

```diff
@@ -196,6 +196,21 @@ impl Keystore for LocalKeystore {
 		self.sign::<ed25519::Pair>(key_type, public, msg)
 	}
 
+	fn with_ed25519_key(
+		&self,
+		key_type: KeyTypeId,
+		public: &ed25519::Public,
+		f: &mut dyn FnMut(&ed25519::Pair),
+	) -> std::result::Result<bool, TraitError> {
```

Comment:
I remember doing a similar function in a branch.

One possible issue is that this is maybe not appropriate for a future hardware keystore, so a specialized function may be more suitable, but at this point I don't know if it is worth doing (given this would mean putting eceis in keystore and I think it was a point that was not suitable from previous PR).

Since it is explicitely a Statement store key that would not be possible to run on HSM, it may not be such an issue.